arcee/cli.py

```diff
@@ -0,0 +1,105 @@
+import os
+from pathlib import Path
+
+import typer
+from typing_extensions import Annotated
+
+from arcee import DALM, upload_doc
+
+cli = typer.Typer()
+"""Arcee CLI"""
+
+
+# model commands; `arcee model [OPTIONS]`
+
+model = typer.Typer(help="Commands to interact with models")
+
+
+@model.command()
+def generate(
+    name: Annotated[str, typer.Option(help="Model name")],
+    query: Annotated[str, typer.Option(help="Query string")],
+    size: Annotated[int, typer.Option(help="Size of the response")] = 3,
+) -> dict:
+    """Generate from model
+
+    Args:
+        name (str): Name of the model
+        query (str): Query string
+        size (int): Size of the response. Defaults to 3.
+
+    """
+
+    dalm = DALM(name=name)
+    return dalm.generate(query=query, size=size)
+
+
+@model.command()
+def retrieve(
+    name: Annotated[str, typer.Option(help="Model name")],
+    query: Annotated[str, typer.Option(help="Query string")],
+    size: Annotated[int, typer.Option(help="Size")] = 3,
+) -> dict:
+    """Retrieve from model
+
+    Args:
+        name (str): Name of the model
+        query (str): Query string
+        size (int): Size of the response. Defaults to 3.
+
+    """
+
+    dalm = DALM(name=name)
+    return dalm.retrieve(query=query, size=size)
+
+
+# command groups; `arcee upload context new_context -f data.jsonl`
+upload = typer.Typer(help="Upload data to Arcee platform")
+
+
+@upload.command()
+def context(
+    name: Annotated[str, typer.Argument(help="Name of the context")],
+    file: Annotated[Path, typer.Option(help="Path to the file", exists=True)],
+) -> None:
+    """Upload a document file to context
+    Args:
+        name (str): Name of the context
+        file (str): Path to the file. Must exist.
+    """
+
+    if not os.path.exists(file):
```

Comment:
```suggestion
    if file.exists():
```

Reply:
Thanks for the comment. I've replaced `os` module with `pathlib` for all validation and read operations.